operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -173,12 +175,51 @@ public FailureCountStep(V1Job domainIntrospectorJob) {
     @Override
     void modifyStatus(DomainStatus domainStatus) {
       domainStatus.incrementIntrospectJobFailureCount(getJobUid());
+      addRetryInfoToStatusMessage(domainStatus);
     }
 
     @Nullable
     private String getJobUid() {
       return Optional.ofNullable(domainIntrospectorJob).map(V1Job::getMetadata).map(V1ObjectMeta::getUid).orElse(null);
     }
+
+    private void addRetryInfoToStatusMessage(DomainStatus domainStatus) {
+
+      if (hasExceededMaxRetryCount(domainStatus)) {
+        domainStatus.setMessage(createStatusMessage(domainStatus, exceededMaxRetryCountErrorMessage()));
+      } else if (isFatalError(domainStatus)) {
+        domainStatus.setMessage(
```

Comment:
The name FATAL_ERROR_DOMAIN_STATUS_MESSAGE doesn't seem to be consistent with the naming we use in MessageKeys. Maybe DOMAIN_FATAL_ERROR would be better, and allow you to dispense with the line break (If not, use a static import on the constant to shorten the line).